### PR TITLE
fix(argo-cd): Added pod exec permission to argo-server Role when exec.enabled is True.

### PR DESCRIPTION
**Pull request by ugoogalizer**
As per issue #1866 , attempting to fix this by adding the relevant code to grant the pod/exec create permission to the argo-server Role when exec.enabled is True.

The argocd-server Role requires the pods/exec create permission in order to be able to start the web based terminal as per: https://argo-cd.readthedocs.io/en/stable/operator-manual/web_based_terminal/

This brings the Role in line with the ClusterRole change already made

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

- [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)

* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

Changes are automatically published when merged to `main`. They are not published on branches.

### CODE REVIEW

File: charts/argo-cd/templates/argocd-server/role.yaml
Comment (mkilchhofer) [inline charts/argo-cd/templates/argocd-server/role.yaml:45]: Do we need an AND condition here to the namespaced / server.clusterAdminAccess option @jmeridth @pdrastil?

**Review by mkilchhofer — COMMENTED** [review]
Hi @ugoogalizer, Thanks for bringing that up and filing a PR. Almost perfect for the first hit 🙏 , but I wanted to ask other maintainers here if we need to check for the "namespaced mode"?

@pdrastil @jmeridth @mbevc1  WDYT?

File: charts/argo-cd/templates/argocd-server/role.yaml
Comment (pdrastil) [reply charts/argo-cd/templates/argocd-server/role.yaml:45]: @mkilchhofer I don't think we need any additional condition for this PR.

The question is if we actually need these checks for Role and ClusterRole as the Argo UI also requires proper permission in Argo RBAC or AppProject. Without these permissions terminal is not shown in the UI.

**Review by pdrastil — APPROVED** [review]

**Comment by mbevc1** [comment]
LGTM if we can get testing passing and resolve conflicts.

**Review by mbevc1 — APPROVED** [review]

File: charts/argo-cd/templates/argocd-server/role.yaml
Comment (mkilchhofer) [reply charts/argo-cd/templates/argocd-server/role.yaml:45]: Agree. Let's merge it :)

**Review by mkilchhofer — APPROVED** [review]

**Comment by ugoogalizer** [comment]
Awesome! Thanks all!

**Comment by ugoogalizer** [comment]
@mkilchhofer  @mbevc1 @pdrastil 

I'm so very sorry, I'm still learning this and my pull request appears to have broken the chart as I've got the yaml spacing wrong somehow. Assuming the best way to fix is to submit another PR request? 

`- apiGroups:
  - ""
resources:
  - pods/exec
verbs:
  - create`

should have been: 

- apiGroups:
  - ""
  resources:
  - pods/exec
  verbs:
  - create

**Comment by pdrastil** [comment]
@ugoogalizer Hi - don't worry this can happen and no one caught this as it's whitespace.

I've opened https://github.com/argoproj/argo-helm/pull/1871 with a fix.

**Pull request merged**
